crates/subspace-node/src/commands/run/consensus.rs

```diff
@@ -397,6 +404,10 @@ pub(super) struct ConsensusChainOptions {
 
     #[clap(flatten)]
     timekeeper_options: TimekeeperOptions,
+
+    /// Experimental support of state-only sync using DSN.
+    #[arg(long, default_value_t = false)]
+    fast_sync: bool,
```

Comment:
I don't think this is the right API, it doesn't scale.

Substrate CLI had this, which we have removed since we only supported full sync:
```
      --sync <SYNC_MODE>
          Blockchain syncing mode.
          
          [default: full]

          Possible values:
          - full:        Full sync. Download end verify all blocks
          - fast:        Download blocks without executing them. Download latest state with proofs
          - fast-unsafe: Download blocks without executing them. Download latest state without proofs
          - warp:        Prove finality and download the latest state
```

I think we should bring `--sync` back and this particular mode can be called "fast" (even though it will be a bit confusing for those farmiliar with Substrate). I'm also not sure it is a good idea to have state pruning defined by a number since IIRC it will prune state regardless of finality, which means node will likely fail to restart in some cases due to state being unavailable at the point of archiver initialization, which is problematic.

This is why I mentioned in the past that we should probably call it "farmer sync" and not support defining state pruning mode explicitly, we should manually prune things when we need to, but not sooner (hence https://github.com/paritytech/polkadot-sdk/issues/1570 from September of last year).

So for now I'd vote for `--sync farmer` and do not allow user to specify `--blocks-pruning` or `--state-pruning` when these are used. Not sure what to do about state pruning yet though. Did you confirm that it doesn't remove non-finalized state?